### PR TITLE
Force to array for groupBy and aggregators in reports

### PR DESCRIPTION
**Pull request by kuzmany**
| Q                                      | A
| -------------------------------------- | ---
| Branch?                                | features
| Bug fix?                               | yes
| New feature?                           | no
| Deprecations?                          | no
| BC breaks?                             | no
| Automated tests included?              | yes
| Related user documentation PR URL      | 
| Related developer documentation PR URL | 
| Issue(s) addressed                     | 

#### Description:
This happend If you use default reports from older instances. If aggregators and orderBy columns are null it return a lot of errors for Leads and Points report (id 5 after installation)

image

Errors:

`[2020-10-26 16:15:40] mautic.WARNING: PHP Warning - array_merge(): Argument #2 is not an array - in file /path/app/bundles/ReportBundle/Entity/Report.php - at line 505 [] [] [2020-10-26 16:15:40] mautic.WARNING: PHP Warning - array_merge(): Argument #1 is not an array - in file /path/app/bundles/ReportBundle/Entity/Report.php - at line 497 [] [] [2020-10-26 16:15:40] mautic.WARNING: PHP Warning - in_array() expects parameter 2 to be array, null given - in file /path/app/bundles/ReportBundle/Event/ReportGeneratorEvent.php - at line 400 {"column":"comp.companyname","columns":null} []`

#### Steps to test this PR:
1. Load up [this PR](https://mautibox.com)
2. Create basic report
3. Force set aggregators and group_by columns to null 
4. See errors in report details reported above
5. After fix it should work properly

### CODE REVIEW

**Review by escopecz — DISMISSED** [review]
Makes sense. But could you also fix the return type comments when it's certain that those 2 methods will always return an array?

File: app/bundles/ReportBundle/Entity/Report.php [hunk 1]
```diff
@@ -451,7 +451,7 @@ public function setGraphs(array $graphs)
      */
     public function getGroupBy()
```
Comment (hluchas) [inline app/bundles/ReportBundle/Entity/Report.php:457]:
```suggestion
    public function getGroupBy(): array
```

File: app/bundles/ReportBundle/Entity/Report.php [hunk 2]
```diff
@@ -466,7 +466,7 @@ public function setGroupBy(array $groupBy)
      */
     public function getAggregators()
```
Comment (hluchas) [inline app/bundles/ReportBundle/Entity/Report.php:472]:
```suggestion
    public function getAggregators(): array
```

**Review by hluchas — CHANGES_REQUESTED** [review]
The return type is always clear here.
Please consider my suggestions
TY!

**Comment by kuzmany** [comment]
@hluchas thanks, make sense. I applied it.

**Review by escopecz — CHANGES_REQUESTED** [review]
As much as I'd like to have the return types it's a BC change. That's why [I suggested to fix only the return types in the comments](https://github.com/mautic/mautic/pull/9392#pullrequestreview-533834795).

Imagine that someone is extending the Report entity and overwriting those 2 methods. It cause a PHP error now as the return type would mismatch.

**Comment by kuzmany** [comment]
@escopecz thanks, I've revert back and change comments type.

**Review by escopecz — DISMISSED** [review]
Thanks @kuzmany! 👍

**Comment by escopecz** [comment]
Although it will need some tests to be mergeable...

**Comment by npracht** [comment]
@escopecz @hluchas can you please review one more time ? I think all your comments have been addressed :)

**Comment by RCheesley** [comment]
@kuzmany I tried to test this but I am unable to replicate the original issue. Is this still relevant?

I have edited the points & leads report in the DB and set the values to (NULL) as you suggest, and when I access the report I am not seeing any errors in the logs (using dev mode).

**Comment by escopecz** [comment]
@RCheesley @npracht https://github.com/mautic/mautic/pull/9392#issuecomment-731996508 still hasn't been addressed

**Comment by kuzmany** [comment]
Close. Do not want to rebase it to M5

**Pull request closed without merging**